File: src/main/java/com/julien/juge/khast/api/KhastApplication.java
package com.julien.juge.khast.api;

import com.julien.juge.khast.api.config.mvc.WebMvcConfiguration;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.PropertySource;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;

@SpringBootApplication
@EnableAutoConfiguration
public class KhastApplication {

	@Bean
	public WebMvcConfigurerAdapter rxJavaWebMvcConfiguration() {
		return new WebMvcConfiguration();
	}

	public static void main(String[] args) {
		SpringApplication.run(KhastApplication.class, args);
	}
}

File: src/main/java/com/julien/juge/khast/api/KhastApplication.kt
package com.julien.juge.khast.api

import com.julien.juge.khast.api.utils.WebSecurityConfig
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean

@SpringBootApplication
@EnableAutoConfiguration
open class KhastApplication {

    @Bean
    open fun rxJavaWebMvcConfiguration(): WebSecurityConfig {
        return WebSecurityConfig()
    }
}

fun main(args: Array<String>) {
    SpringApplication.run(KhastApplication::class.java, *args)
}